cmd/errtrace/main_test.go

```diff
@@ -54,25 +54,51 @@ func TestGolden(t *testing.T) {
 	for _, file := range files {
 		name := strings.TrimSuffix(filepath.Base(file), ".go")
 		t.Run(name, func(t *testing.T) {
-			testGolden(t, file)
+			testGoldenFile(t, file)
 		})
 	}
 }
 
-func testGolden(t *testing.T, file string) {
+func testGoldenFile(t *testing.T, file string) {
 	giveSrc, err := os.ReadFile(file)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	wantLogs, err := extractLogs(giveSrc)
+	wantSrc, err := os.ReadFile(file + ".golden")
 	if err != nil {
-		t.Fatal(err)
+		t.Fatal("Bad test: missing .golden file:", err)
 	}
 
-	wantSrc, err := os.ReadFile(file + ".golden")
+	type runTests struct {
+		noOptions  bool
+		optNoWrapN bool
+	}
+	run := runTests{true, true} // by default, run all tests.
+	if strings.Contains(string(giveSrc), "@runIf options=<empty>") {
+		run = runTests{noOptions: true}
+	}
+	if strings.Contains(string(giveSrc), "@runIf options=no-wrapn") {
+		run = runTests{optNoWrapN: true}
```

Comment:
Heh, I did imagine us getting to having to specify options in the comments at the top of the file at some point.